specs/eventing/control-plane.md

```diff
@@ -0,0 +1,1120 @@
+   </td>
+   <td>Timestamp
+   </td>
+   <td>Last update time for this condition.
+   </td>
+   <td>"" – may be unset
+   </td>
+  </tr>
+</table>
+
+Additionally, the resource's `status.conditions` field MUST be managed as
+follows to enable clients (particularly user interfaces) to present useful
+diagnostic and error message to the user. In the following section, conditions
+are referred to by their `type` (aka the string value of the `type` field on the
+Condition).
+
+1.  Each resource MUST have either a `Ready` condition (for ongoing systems) or
+    `Succeeded` condition (for resources that run to completion) with
+    `severity=""`, which MUST use the `True`, `False`, and `Unknown` status
+    values as follows:
+
+    1.  `False` MUST indicate a failure condition.
+    1.  `Unknown` SHOULD indicate that reconciliation is not yet complete and
+        success or failure is not yet determined.
+    1.  `True` SHOULD indicate that the application is fully reconciled and
+        operating correctly.
+
+    `Unknown` and `True` are specified as SHOULD rather than MUST requirements
+    because there may be errors which prevent functioning which cannot be
+    determined by the API stack (e.g. DNS record configuration in certain
+    environments). Implementations are expected to treat these as "MUST" for
+    factors within the control of the implementation.
+
+1.  For non-`Ready` conditions, any conditions with `severity=""` (aka "Error
+    conditions") must be aggregated into the "Ready" condition as follows:
+
+    1.  If the condition is `False`, `Ready` MUST be `False`.
+    1.  If the condition is `Unknown`, `Ready` MUST be `False` or `Unknown`.
+    1.  If the condition is `True`, `Ready` may be any of `True`, `False`, or
+        `Unknown`.
+
+    Implementations MAY choose to report that `Ready` is `False` or `Unknown`
+    even if all Error conditions report a status of `True` (i.e. there may be
+    additional hidden implementation conditions which feed into the `Ready`
+    condition which are not reported.)
+
+1.  Non-`Ready` conditions with non-error severity MAY be surfaced by the
+    implementation. Examples of `Warning` or `Info` conditions could include:
+    missing health check definitions, scale-to-zero status, or non-fatal
+    capacity limits.
+
+Conditions type names should be chosen to describe positive conditions where
+`True` means that the condition has been satisfied. Some conditions may be
+transient (for example, `ResourcesAllocated` might change between `True` and
+`False` as an application scales to and from zero). It is RECOMMENDED that
+transient conditions be indicated with a `severity="Info"`.
+
+## Resource Lifecycle
+
+### Broker Lifecycle
+
+A Broker represents an Addressable endpoint (i.e. it has a `status.address.url`
+field) which can receive, store, and forward events to multiple recipients based
+on a set of attribute filters (Triggers). Triggers are associated with a Broker
```

Comment:
Triggers MUST be associated with a Broker via its `spec.Broker` field....

Reply:
Do we need to mandate that they're both in the same namespace? Or do we want to be future proof and allow for cross NS stuff?

Reply:
At the moment, we don't have a field to indicate the namespace of the Broker. I think the assumption is that this would be the same namespace as the Trigger, but if you wanted to have a canonical namespace that contained all the Brokers for the cluster (vis-a-vis the Code Engine discussion), this text would allow that.

Reply:
I think I'd want to carefully consider the access control implications before allowing cross-namespace triggers, so I think for this version of the spec, we shouldn't include a namespace field on the Trigger.